Client/Application/AppDelegate.swift

```diff
@@ -483,11 +483,10 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UIViewControllerRestorati
     }
 
     fileprivate func openURLsInNewTabs(_ notification: UNNotification) {
-        if let sentURLs = notification.request.content.userInfo["sentTabs"] as? [NSDictionary] {
-            for sentURL in sentURLs {
-                if let urlString = sentURL.value(forKey: "url") as? String, let url = URL(string: urlString) {
-                    browserViewController.openURLInNewTab(url, isPrivileged: true)
-                }
+        guard let urls = notification.request.content.userInfo["sentTabs"] as? [NSDictionary]  else { return }
+        for sentURL in urls {
+            if let urlString = sentURL.value(forKey: "url") as? String, let url = URL(string: urlString) {
+                receivedURLs.append(url)
```

Comment:
This is the critical fix. Add the URL to the receivedURLs list, which is safely processed later. The rest of the code is changing the receivedURLs list to be accessible here.